### DISCUSSION TITLE
custom CSS?

### DISCUSSION THREAD

**Discussion by publicityreform** [knowledgefutures/pubpub#1307] [Support]
just curious if there is a way (or a plan) to more easily customize the CSS for a community, beyond the basic settings? such as as the ability to add / edit an external stylesheet? 

specifically thinking about having more control over how auto-generated pdf's look, as well as the default styles for media and embedded content across all of our community's pubs. 

would this require hosting our own instance of pubpub?

thank you!

**Comment by gabestein** [comment]
Thanks so much for the question. At the moment, the only way to really do CSS customization is to hack it by adding CSS to HTML blocks on pages. This works fine, and you can see some very cool examples like [Sonic Scope](https://www.sonicscope.org/) and our own [Commonplace](https://commonplace.knowledgefutures.org), but we only recommend it if you can maintain it, as the classes are subject to change over time.

More generally, we have received increasing requests for customization, and are considering ways we can provide more capabilities there without sacrificing the ease-of-use and relatively foolproof customization that's one of our key features. There are also some architectural concerns involving supporting custom CSS within the real-time editor, which in some instances relies on specific styles to function properly. If you could provide us with some more detail about what you're looking to accomplish, that would help us better understand your needs and think about how we might be able to incorporate them into our customization feature ideas. Feel free to reply here or send a note to hello@pubpub.org.

On the PDF export front, we'll definitely offer more options at some point down the road, including potentially complete customization as part of some of the higher end packages. It's a little easier architecturally to make customization here possible, but at the same time, because we're running a little on the bleeding edge using Paged Media to generate PDFs, there are a few more pitfalls. Similarly, knowing more about what you're looking to do would help.

We'd love to be able to recommend firing up your own instances of PubPub to do some of this customization, but in all honesty, our focus right now with our limited resources is on supporting the hosted version, and so we don't really recommend it unless you're very technical and familiar with most parts of our stack.

**Comment by ryoungblom** [comment]
Not 100% sure what you're going for as far as custom PDF styling, and this definitely doesn't solve your larger question, but just to add a possible workaround specifically on the PDF front: In the "pub settings" for each pub, there is an option to upload a custom PDF that will be used as that pub's default download. If (at least part of your issue is) you're just hoping to offer PDF downloads that are structured differently than the auto-generated defaults, customizing a PDF elsewhere and then uploading this custom PDF to serve as the default download may work as a short-term solution.
